server/raft_transport.go

```diff
@@ -32,8 +33,19 @@ import (
 const (
 	raftServiceName = "MultiRaft"
 	raftMessageName = raftServiceName + ".RaftMessage"
+	// Outgoing messages are queued on a per-node basis on a channel of
+	// this size.
+	raftSendBufferSize = 500
+	// When no message has been sent to a Node for that duration, the
+	// corresponding instance of processQueue will shut down.
+	raftIdleTimeout = time.Minute
```

Comment:
I'm not sure an idle timeout makes sense. It's cheap to start and stop goroutines so we could stop the processQueue thread as soon as the queue becomes empty. Conversely, it should be affordable to simply keep one processQueue goroutine per peer node for the life of the transport. Running goroutines are cheap and we expect most node-to-node connections to be active most of the time.

I'm concerned about race conditions when a new outgoing message appears just as the old goroutine is shutting down; with an idle timeout this race will occur rarely and be difficult to test. If we shut down the goroutine eagerly we are much more likely to uncover all the bugs; if we never shut it down we avoid the race completely.